### PR TITLE
loop-util tweaks

### PR DESCRIPTION
**Pull request by poettering**
The safe subset of #22992, split out.

Let' merge these fixes first, they make sense on their own.

### CODE REVIEW

File: src/test/test-loop-block.c [hunk 1]
```diff
@@ -228,19 +228,27 @@ int main(int argc, char *argv[]) {
                        slow_tests_enabled() ? 5 * USEC_PER_SEC :
                        1 * USEC_PER_SEC);
 
-        for (unsigned i = 0; i < N_THREADS; i++)
-                assert_se(pthread_create(threads + i, NULL, thread_func, FD_TO_PTR(fd)) == 0);
+        assert_cc(N_THREADS > 0);
+
+        if (N_THREADS > 1) {
```
Comment (bluca) [inline src/test/test-loop-block.c]: single line if -> no brackets

**Review by bluca — APPROVED** [review]

**Review by yuwata — APPROVED** [review]

**Comment by bluca** [comment]
```
../src/test/test-loop-block.c:32:14: error: unused function 'thread_func' [-Werror,-Wunused-function]
static void* thread_func(void *ptr) {
```

**Comment by poettering** [comment]
> ../src/test/test-loop-block.c:32:14: error: unused function 'thread_func' [-Werror,-Wunused-function]
> static void* thread_func(void *ptr) {

fixed

**Comment by bluca** [comment]
```
../src/test/test-loop-block.c:30:15: error: unused variable 'end' [-Werror,-Wunused-variable]
static usec_t end = 0;
```

**Pull request merged**